images/agent/src/pkg/controller/block_device.go

```diff
@@ -162,7 +162,30 @@ func hasBlockDeviceDiff(blockDevice v1alpha1.BlockDevice, candidate internal.Blo
 		!reflect.DeepEqual(ConfigureBlockDeviceLabels(blockDevice), blockDevice.Labels)
 }
 
-func GetAPIBlockDevices(ctx context.Context, kc kclient.Client, metrics monitoring.Metrics) (map[string]v1alpha1.BlockDevice, error) {
+func GetAPIBlockDevicesBySelector(ctx context.Context, cl client.Client, metrics monitoring.Metrics, selector *metav1.LabelSelector) (map[string]v1alpha1.BlockDevice, error) {
+	list := &v1alpha1.BlockDeviceList{}
+	s, err := metav1.LabelSelectorAsSelector(selector)
+	if err != nil {
+		return nil, err
+	}
+	start := time.Now()
+	err = cl.List(ctx, list, &client.ListOptions{LabelSelector: s})
+	metrics.APIMethodsDuration(BlockDeviceCtrlName, "list").Observe(metrics.GetEstimatedTimeInSeconds(start))
+	metrics.APIMethodsExecutionCount(BlockDeviceCtrlName, "list").Inc()
+	if err != nil {
+		metrics.APIMethodsErrors(BlockDeviceCtrlName, "list").Inc()
+		return nil, err
+	}
+
+	result := make(map[string]v1alpha1.BlockDevice, len(list.Items))
+	for _, i := range list.Items {
+		result[i.Name] = i
+	}
+
+	return result, nil
+}
+
+func GetAPIBlockDevices(ctx context.Context, kc client.Client, metrics monitoring.Metrics) (map[string]v1alpha1.BlockDevice, error) {
```

Comment:
This method is mostly the same as `GetAPIBlockDevicesBySelector`, we can refactor it into using `GetAPIBlockDevicesBySelector` with `nil` `selector`.